### PR TITLE
Allow custom groups to be added to APIM

### PR DESCRIPTION
**Pull request by EwanNoble**

### CODE REVIEW

File: templates/apim/apim-group.json [hunk 1]
```diff
@@ -18,17 +20,19 @@
...
     }
   },
-  "variables": {},
+  "variables": {
+    "isExternal": "[and(greater(length(parameters('tenantId')) ,0), greater(length(parameters('aadGroupObjectId')), 0))]"
+  },
   "resources": [
     {
-      "name": "[concat(parameters('apimName'), '/', parameters('aadGroupObjectId'))]",
+      "name": "[concat(parameters('apimName'), '/', if(variables('isExternal'), parameters('aadGroupObjectId'), parameters('groupDisplayName')))]",
```
Comment (jack-cloud-platform) [inline templates/apim/apim-group.json:28]:
```suggestion
      "name": "[concat(parameters('apimName'), '/', if(variables('isExternal'), parameters('groupDisplayName'), parameters('aadGroupObjectId')))]",
```
Should the order for the if statement be the other way round?

File: templates/apim/apim-group.json [hunk 2]
```diff
@@ -18,17 +20,19 @@
       "type": "string"
     }
   },
-  "variables": {},
+  "variables": {
+    "isExternal": "[and(greater(length(parameters('tenantId')) ,0), greater(length(parameters('aadGroupObjectId')), 0))]"
```
Comment (jack-cloud-platform) [inline templates/apim/apim-group.json:24]:
```suggestion
    "isExternal": "[not(and(greater(length(parameters('tenantId')) ,0), greater(length(parameters('aadGroupObjectId')), 0)))]"
```

File: templates/apim/apim-group.json [hunk 3]
```diff
@@ -18,17 +20,19 @@
...
+      "name": "[concat(parameters('apimName'), '/', if(variables('isExternal'), parameters('aadGroupObjectId'), parameters('groupDisplayName')))]",
       "type": "Microsoft.ApiManagement/service/groups",
       "apiVersion": "2019-12-01",
       "properties": {
         "displayName": "[parameters('groupDisplayName')]",
         "description": "[parameters('groupDescription')]",
-        "type": "external",
-        "externalId": "[concat('aad://', parameters('tenantId'), '/groups/', parameters('aadGroupObjectId'))]"
+        "type": "[if(variables('isExternal'), 'external', 'custom')]",
+        "externalId": "[if(variables('isExternal'), concat('aad://', parameters('tenantId'), '/groups/', parameters('aadGroupObjectId')), json('null'))]"
```
Comment (jack-cloud-platform) [inline templates/apim/apim-group.json:35]:
```suggestion
        "externalId": "[if(variables('isExternal'), concat('aad://', parameters('tenantId'), '/groups/', parameters('aadGroupObjectId')), json('null'))]"
```
I think order for if needs to change here as well

File: templates/apim/apim-group.json (hunk 2 above)
Comment (EwanNoble) [reply templates/apim/apim-group.json:24]: External is the term used by APIM for AAD or external identity provider groups (look at the `type` property) , therefore if an AAD group is specified it `isExternal`

**Review by jack-cloud-platform — APPROVED** [review]

**Pull request merged**